### PR TITLE
[ART-2353] Clean metadata advisory before attaching builds

### PR DESCRIPTION
**Pull request by thiagoalessio**
To avoid a mix of `.stage` and `.prod` builds in the same advisory.

### CODE REVIEW

**Review by joepvd — CHANGES_REQUESTED** [review]
If only a single operator needs to get rebuilt and attached, now all others will get removed. Not sure that's the behavior I'd expect.

File: jobs/build/appregistry/appregistry.groovy [hunk 1]
```diff
@@ -348,6 +352,7 @@ def stageAttachMetadata(operatorBuilds) {
         ${params.DRY_RUN ? "--noop" : ""}
     """
 
+    cleanAdvisoryBuilds(params.METADATA_ADVISORY)
```
Comment (vfreex) [inline jobs/build/appregistry/appregistry.groovy:355]: As @joepvd mentioned, if we only rebuild specific metadatas, this is going to remove all metadata builds. Would like to have a smarter way (e.g. filter by environment name then remove them with elliott)

**Review by vfreex — CHANGES_REQUESTED** [review]

File: jobs/build/appregistry/appregistry.groovy (hunk 1 above)
Comment (vfreex) [reply jobs/build/appregistry/appregistry.groovy:355]: Or a simple approach: add an option to opt out this behavior if checked.

**Comment by thiagoalessio** [comment]
Thanks @vfreex and @joepvd for the review! I changed the strategy: I list all currently attached builds, and remove _only_ the ones that don't have the current `STREAM` on its NVR.
So, if we are running it with `STREAM=prod`, it would remove all builds that have no `".prod"` on the NVR.

**Review by sosiouxme — APPROVED** [review]
nice

**Pull request merged**